core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -499,6 +499,25 @@ class SnappyConf(@transient val session: SnappySession)
           pool.toString
         case Some(pool) => throw new IllegalArgumentException(s"Invalid Pool $pool")
       }
+
+    case Property.PlanCaching.name =>
+      value match {
+        case Some(boolval) => {
+          if (boolval.toString.toBoolean) session.clearPlanCache()
+        }
```

Comment:
As mentioned, the planCaching should be made a var instead of def and that should then be updated here.

Reply:
ok.